source/Mlos.Model.Services.Client/OptimizerServiceEncoderDecoder.cs

```diff
@@ -0,0 +1,419 @@
+                ChunkType = DimensionToGrpcType(dimension.ChunkType),
+                Chunks = { chunks },
+            };
+        }
+
+        public OptimizerService.DiscreteDimension EncodeDiscreteDimension(DiscreteDimension dimension)
+        {
+            return new OptimizerService.DiscreteDimension
+            {
+                Name = dimension.Name,
+                Min = dimension.Min,
+                Max = dimension.Max,
+            };
+        }
+
+        public OptimizerService.EmptyDimension EncodeEmptyDimension(EmptyDimension dimension)
+        {
+            return new OptimizerService.EmptyDimension
+            {
+                Name = dimension.Name,
+                DimensionType = DimensionToGrpcType(dimension.ObjectType),
+            };
+        }
+
+        public OptimizerService.ContinuousDimension EncodeContinuousDimension(ContinuousDimension dimension)
+        {
+            return new OptimizerService.ContinuousDimension
+            {
+                Name = dimension.Name,
+                Min = dimension.Min,
+                Max = dimension.Max,
+                IncludeMin = dimension.IncludeMin,
+                IncludeMax = dimension.IncludeMax,
+            };
+        }
+
+        public OptimizerService.OrdinalDimension EncodeOrdinalDimension(OrdinalDimension dimension)
+        {
+            var instance = new OptimizerService.OrdinalDimension();
+            instance.Name = dimension.Name;
+            instance.Ascending = dimension.Ascending;
+            foreach (var value in dimension.OrderedValues)
+            {
+                instance.OrderedValues.Add(EncodePrimitiveValue(value));
+            }
+
+            return instance;
+        }
+
+        public OptimizerService.CategoricalDimension EncodeCategoricalDimension(CategoricalDimension dimension)
+        {
+            var instance = new OptimizerService.CategoricalDimension();
+            instance.Name = dimension.Name;
+            foreach (var value in dimension.Values)
+            {
+                instance.Values.Add(EncodePrimitiveValue(value));
+            }
+
+            return instance;
+        }
+
+        public OptimizerService.PrimitiveValue EncodePrimitiveValue(object value)
+        {
+            var data = new OptimizerService.PrimitiveValue();
+            if (value is int)
+            {
+                data.IntValue = (int)value;
+            }
+            else if (value is bool)
+            {
+                data.BoolValue = (bool)value;
+            }
+            else if (value is string)
+            {
+                data.StringValue = (string)value;
+            }
+            else if (value is double)
+            {
+                data.DoubleValue = (double)value;
+            }
+            else
+            {
+                System.Console.Error.WriteLine("Invalid type provided to OptimizerServiceEncoderDecoder.Encoder.EncodePrimitiveValue(..). Type: " + value.GetType());

```

Comment:
throw Exception
DO NOT LOG TO CONSOLE

Reply:
https://github.com/zaporter/MLOS/commit/d30acf1866fc84d8d2f164101513181df4abfb36